source/common/http/http_server_properties_cache_impl.cc

```diff
@@ -296,6 +297,30 @@ HttpServerPropertiesCacheImpl::getOrCreateHttp3StatusTracker(const Origin& origi
   return *it->second.h3_status_tracker;
 }
 
+bool HttpServerPropertiesCacheImpl::isHttp3Broken(const Origin& origin) {
+  absl::optional<Origin> canonical = getCanonicalOrigin(origin.hostname_);
+
+  if (!Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.use_canonical_suffix_for_quic_broken") ||
+      !canonical.has_value()) {
+    return getOrCreateHttp3StatusTracker(origin).isHttp3Broken();
+  }
+
+  // Note that we don't create a new tracker for the origin if there's already a corresponding one
+  // for the canonical origin.
+  auto entry_it = protocols_.find(origin);
```

Comment:
nit: You can use if expression, i.e. `if (auto entry_it = protocols_.find(origin); ....) { ... }`